### ISSUE TITLE
Need Readonly PDF

### ISSUE THREAD

**Issue by johndave8116** [codeshell/fpdm#2]
Hello Team,

Your code and this library is perfect and awesome - I really liked it.
But I have only one concern - once the data is merged into the PDF - I want the final PDF but not in Editable mode.
Is it possible? If Yes then please let me know how.
Really Appreciate your earliest reply.

Thank you.

**Comment by judodan** [comment]
I was looking for the very same thing. Currently I am using FPDI to do this, but the problem is that when you change the form, you have to painstakingly go through the coordinates. Being able just to pass an array and have it produce an uneditable file would be great.

**Comment by mgip** [comment]
Same problem here, some solution for that?

**Comment by imclean557** [comment]
All that's needed is a flag added to the field: `  /Ff 1` . See [PDF Reference](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/pdf_reference_archives/PDFReference.pdf)

There's a lot of repeated code for calculating the offset which could probably be consolidated into a single method. But for this feature, would it be enough to be able to pass a "read_only" option when merging or setting a field's value?

**Comment by imclean557** [comment]
#31

**Comment by imclean557** [comment]
Adding support in the linked PR might help move things along. Does it work for you?

**Comment by BeRightBaack** [comment]
I was able to accomplish this simply by using readonly pdf form.
Using something like https://www.pdfescape.com you can edit your form, right click on every field and in setting set it as read-only. If you use that template, values will then be added in it and will not be editable.
OFC in my case i had to run template through pdftk before I could use it with script.

Don't know your exact usage and process, but give it a try, maybe it will work for you too
![s181](https://user-images.githubusercontent.com/28875429/146774071-9e01d051-b886-4f41-8ab7-0f565e495784.png)
